universal-application-tool-0.0.1/test/repository/ApplicationRepositoryTest.java

```diff
@@ -49,6 +50,10 @@ public void submitApplication_updatesOtherApplications() {
     assertThat(
             repo.getApplication(appOne.id).toCompletableFuture().join().get().getLifecycleStage())
         .isEqualTo(LifecycleStage.OBSOLETE);
+    // Ensure the second submitted app has a later submit time than the first.
+    assertThat(appThree.getSubmitTime()).isAfter(appOne.getSubmitTime());
+    // Ensure the submit time didn't get changed when the application was set as obsolete.
+    assertThat(appOne.getSubmitTime()).isEqualTo(initialSubmitTime);
```

Comment:
We should have a test case to cover that a draft/not submitted application does not have a submit time.

Can we add a check to ensure that appTwo does not have a submit time?

Reply:
Idk why this comment showed up twice lol sorry.

Reply:
Haha all good! Added null check.